analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodWalker.cs

```diff
@@ -61,16 +61,22 @@ public override void VisitThrowStatement(ThrowStatementSyntax node)
             {
```

Comment:
This whole class is actually a helper and not a rule, so I suggest moving it to helpers.